src/main/osd/osd.c

```diff
@@ -769,6 +770,15 @@ static bool osdDisplayStat(int statistic, uint8_t displayRow)
             return true;
         }
         break;
+    
+    case OSD_STAT_WATT_HOURS_DRAWN:
+        if (batteryConfig()->currentMeterSource != CURRENT_METER_NONE) {
+            const float wattHoursDrawn = getMAhDrawn() * getBatteryVoltage() / 100000.0f;
```

Comment:
This doesn't seem correct to me. More like an estimation after one flight. How did you derive this line of code?
According to my derivation, the formula for precise Wh values at every point in time is
```
wattHoursDrawn += getBatteryVoltage() * (getMAhDrawn() - mAhDrawnPrev) / 10.0f;
```

Reply:
I see, so I was looking at it for a single flight or as long as the battery isn't unplugged this is the value. It is just converting the mAh drawn to Wh based on voltage. Did I interpret the feature request wrong?

Reply:
The problem is a mathematical one. Voltage changes at every point in time, so you cannot just use an arbitrary voltage from the end of the flight and multiply it with the mah used, because your value will be off. You need to integrate the electrical power used at every iteration to get the energy used.

Reply:
I see so at first for a 4s pack it would be 1680 for the voltage but by the end it would result in 1475, this is basically just using that end value. That is definitely the result I was seeing. Okay I will correct this. I believe the osd element will be wrong as well then, going to update.

Reply:
Also, voltage is changing a lot when you hit the throttle. Best way is if you let BF sum up the Wh used somewhere in battery.c and then you just call a function `getWhUsed()` to get the number whenever you need it.